### PR TITLE
feat: Add Go SDK custom header support and README

### PR DESCRIPTION
**Pull request by jeremytchang**
Following our node sdk's design, allow developers to set custom headers on Go SDK requests. Request specific custom headers will override headers defined in the sdk's settings/config. This will allow the Go SDK caller to set `Accept` headers and other custom headers. Also add README for Go SDK.

Also fixes issue stated in https://github.com/looker-open-source/sdk-codegen/issues/1075. (Not the main one, but other one brought up by Go SDK users)

### CODE REVIEW

File: go/rtl/auth_test.go [hunk 1]
```diff
@@ -125,70 +125,6 @@ func TestAuthSession_Do_Authorization(t *testing.T) {
 	})
 }
 
-func TestAuthSession_Do_UserAgent(t *testing.T) {
```
Comment (jeremytchang) [inline go/rtl/auth_test.go:128]: Moved these tests to header section.

File: go/rtl/auth.go [hunk 2]
```diff
@@ -142,7 +142,7 @@ func (s *AuthSession) Do(result interface{}, method, ver, path string, reqPars m
 	}
 
 	// set headers
-	req.Header.Add("Content-Type", contentTypeHeader)
+	req.Header.Set("Content-Type", contentTypeHeader)
 
 	if s.Config.AgentTag != "" {
 		req.Header.Set("User-Agent", s.Config.AgentTag)
```
Comment (jkaster) [inline go/rtl/auth.go:148]: you should also set `x-looker-appid` like the other SDKs do. There are some request routers that will modify the `User-Agent` header. Currently this is set in the existing `auth.go` and the intent is that AgentTag overwrites that default header value.

File: go/README.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    // token from your Looker instance's `login` endpoint. 
+    session := rtl.NewAuthSession(cfg)
+
+    // Create new instance of the Go Looker SDK 
+	sdk := v4.NewLookerSDK(session)
+
+    // Call the Looker API e.g. get your user's name
+    me, err := sdk.Me("", nil)
+	fmt.Printf("Your name is %s %s\n", *(me.first), *(me.last))
+}
```
Comment (jkaster) [inline go/README.md:34]: code alignment seems strange here

File: go/README.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        "HeaderName2": "HeaderValue2",
+    }})
+}
+```
+
+### Timeout 
+
+You can set a custom timeout (in seconds) on Looker Go SDK's requests. The timeout defaults to 120 seconds. A timeout can either be applied to all outgoing requests or per outgoing request. 
+
+#### Timeout for all requests
```
Comment (jkaster) [inline go/README.md:84]: great addition!

File: go/README.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        "HeaderName1": "HeaderValue1",
+        "HeaderName2": "HeaderValue2",
+    }
+    
+    session := rtl.NewAuthSession(cfg)
+	sdk := v4.NewLookerSDK(session)
+}
+```
+
+#### Custom headers per request
```
Comment (jkaster) [inline go/README.md:62]: This will definitely be handy in some use cases

File: go/README.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+	cfg, err := rtl.NewSettingsFromFile("path/to/looker.ini", nil)
+    session := rtl.NewAuthSession(cfg)
+	sdk := v4.NewLookerSDK(session)
+
+    // Set the timeout in the options passed into the SDK method
+    me, err := sdk.Me("", &ApiSettings{Timeout: 60})
+
+    if errors.Is(err, context.DeadlineExceeded) {
+        // Timeout exceeded
+    }
```
Comment (jkaster) [inline go/README.md:116]: strange indentation again? is it mixed tabs and spaces?

**Review by jkaster — CHANGES_REQUESTED** [review]
This is going to be some appreciated new functionality

File: go/rtl/auth.go (hunk 2 above)
Comment (jeremytchang) [reply go/rtl/auth.go:148]: Good callout, was unaware of the behavior of AgentTag in relation to `x-looker-appid`. Will have a separate ticket to tackle that.

File: go/README.md (hunk 6 above)
Comment (jeremytchang) [reply go/README.md]: Yup mixed by accident. Fixed!

File: go/README.md (hunk 3 above)
Comment (jeremytchang) [reply go/README.md]: Fixed!

**Review by jkaster — APPROVED** [review]
LGTM with the follow-up ticket

**Pull request merged**